graphstorm-processing/scripts/run_distributed_processing.py

```diff
@@ -0,0 +1,265 @@
+    raw_output = args.s3_output_prefix
+    s3_input_prefix = raw_input[:-1] if raw_input[-1] == "/" else raw_input
+    s3_output_prefix = raw_output[:-1] if raw_output[-1] == "/" else raw_output
+    s3_train_config = f"{s3_input_prefix}/{args.config_filename}"
+
+    code_path = args.code_path if args.code_path else Path(_ROOT).parent / "graphstorm_processing"
+
+    if args.job_name is None:
+        args.job_name = "gs-processing"
+
+    instance_count = args.instance_count
+    timestamp = strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+    processing_job_name = f"{args.job_name}-pyspark-{timestamp}"
+    if len(processing_job_name) > 63:
+        # Timestamp will be 19 chars, so we take 43 chars from the prefix plus hyphen
+        processing_job_name = f"{args.job_name[:43]}-{timestamp}"
+
+    s3_train_config = (
+        f"{s3_input_prefix}{args.config_filename}"
+        if s3_input_prefix.endswith("/")
+        else f"{s3_input_prefix}/{args.config_filename}"
+    )
+    s3_input_bucket = s3_input_prefix.split("/")[2]
+    s3_input_key = s3_input_prefix.split("/", 3)[3]
+
+    if not s3_output_prefix:
+        s3_bucket = s3_input_prefix.split("/")[2]
+        s3_output_prefix = (
+            f"s3://{s3_bucket}/graphstorm-distributed-graph-processing/output/{processing_job_name}"
+        )
+
+    s3_output_bucket = s3_output_prefix.split("/")[2]
+    print(
+        "Creating SageMaker Processing job for train"
+        f" config in {s3_train_config} and outputs to {s3_output_prefix}"
+    )
+
+    # Create SageMaker session with high retry count to avoid thresholding errors
+    args.region = (
+        script_utils.get_bucket_region(s3_output_bucket) if not args.region else args.region
+    )
+    sm_boto, s3_boto = [
+        boto3.client(
+            service,
+            config=Config(
+                region_name=args.region,
+                connect_timeout=5,
+                read_timeout=60,
+                retries={"max_attempts": 20},
+            ),
+            region_name=args.region,
+        )
+        for service in ["sagemaker", "s3"]
+    ]
+    sagemaker_session = sagemaker.Session(
+        boto_session=boto3.Session(region_name=args.region), sagemaker_client=sm_boto
+    )
+
+    processor_kwargs = script_utils.parse_processor_kwargs(args.sm_processor_parameters)
+
+    byte_size_on_s3 = script_utils.determine_byte_size_on_s3(
+        s3_input_bucket, s3_input_key, s3_boto
+    ) // (1024 * 1024 * 1024)
+    input_total_size_in_gb = max(1, byte_size_on_s3 // (1024 * 1024 * 1024))
+    # Heuristic: total storage of 6+ times the total input size should be
+    # sufficient for Spark (assuming CSV input)
+    # Using anything less than 2*input_size/instance_count led to failures for large datasets
+    max_allowed_volume_size = script_utils.get_max_volume_size_for_processing(args.region)
+    if "volume_size_in_gb" not in processor_kwargs:
+        desired_volume_size = 6 * (input_total_size_in_gb // instance_count)
```

Comment:
Why 6X?

Reply:
This is empirical and a "magic" number. I've often observed running out of disk space when trying to use 2-4x, so this is a "safe" value to use.